### PR TITLE
CI job running tests with queuing on

### PR DESCRIPTION
**Pull request by gjoseph92**
Adds a CI job, only for Ubuntu and py3.9, running the test suite with `worker-saturation: 1.0` (instead of `inf`).

Also adds a `@pytest.mark.oversaturate_only` marker, which automatically skips marked tests if queuing is enabled.

A surprisingly small number of tests actually failed when queuing was turned on. I've updated the ones that did as needed. I've tried as much as possible to just make the tests agnostic to queuing. In a few cases though, I've needed to add explicit conditional logic. And in a couple cases, I've marked tests to be skipped when queuing is enabled, since they may not make sense.

Unfortunately there are three tests that are flaky under queuing, which could indicate an actual bug. They're skipped for now, but should be investigated.

Closes https://github.com/dask/distributed/issues/6631

- [x] Tests added / passed
- [x] Passes `pre-commit run --all-files`

### CODE REVIEW

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -107,6 +116,11 @@ jobs:
 
           python continuous_integration/scripts/host_info.py
 
+      - name: Enable queuing
+        if: ${{ matrix.queuing == 'queue' }}
+        shell: bash -l {0}
+        run: export DASK_DISTRIBUTED__SCHEDULER__WORKER_SATURATION=1.0
+
```
Comment (crusaderky) [inline .github/workflows/tests.yaml:123]: I thought that environment variables are reset at every step of the github action?

**Review by crusaderky — CHANGES_REQUESTED** [review]
Instead of this new `oversaturate_only` mark, couldn't you simply have

```python
OVERSATURATION = math.isinf(dask.config.get("distributed.scheduler.worker-saturation"))
@pytest.mark.flaky(not OVERSATURATION, reason="flaky on MacOS")
```
Tests that don't make sense with queueing should instead force 
```python
config={"distributed.scheduler.worker-saturation": math.inf}
```
(some already do)

File: .github/workflows/tests.yaml [hunk 2]
```diff
@@ -44,8 +53,8 @@ jobs:
         # run: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20]
 
     env:
-      TEST_ID: ${{ matrix.os }}-${{ matrix.python-version }}-${{ matrix.partition-label }}
-      # TEST_ID: ${{ matrix.os }}-${{ matrix.python-version }}-${{ matrix.partition-label }}-${{ matrix.run }}
+      TEST_ID: ${{ matrix.os }}-${{ matrix.python-version }}-${{ matrix.partition-label }}-${{ matrix.queuing }}
```
Comment (fjetter) [inline .github/workflows/tests.yaml:56]: I think this renaming will break some logic in our test report generation.

https://github.com/dask/distributed/blob/acf607832c7191cc496a9b4a81760170de85062c/continuous_integration/scripts/test_report.py#L146-L152
https://github.com/dask/distributed/blob/acf607832c7191cc496a9b4a81760170de85062c/continuous_integration/scripts/test_report.py#L353-L357

I introduced this in https://github.com/dask/distributed/pull/6837 

Worst case, we remove the more specific job URL again (we knew from the start this is very brittle)

Comment (fjetter) [reply .github/workflows/tests.yaml:56]: see https://github.com/dask/distributed/pull/6837/files#r939182529

Comment (fjetter) [reply .github/workflows/tests.yaml:56]: I don't think this should block this PR

**Comment by gjoseph92** [comment]
* Removed the `oversaturate_only` mark; just using `pytet.mark.skipif` and `config={"distributed.scheduler.worker-saturation": math.inf}` as necessary
* Fixed setting the environment variable in the test step
* Updated test report script to handle and display results from the `queue` job:

    visualization

Ready for re-review and merge.

**Comment by gjoseph92** [comment]
- https://github.com/dask/distributed/issues/6999

**Review by crusaderky — APPROVED** [review]

**Comment by crusaderky** [comment]
Thank you!

**Comment by jrbourbeau** [comment]
Just to confirm, is the new CI job added here for testing purposes and is planned to be removed in the future, or is this intended as a long-term change?

**Comment by crusaderky** [comment]
Long term, we'll want to enable scheduler-side queueing by default.
When that happens, we'll need to revisit the unit tests to have selected few tests with queueing off (like it already happens with all other config toggles) and revert this PR.
Before that, all tests that are flaky only when queueing is on need to be addressed.

**Pull request merged**